sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -161,6 +161,267 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession with AdaptiveSpark
     }
   }
 
+  private[this] def hexToBytes(s: String): Array[Byte] = {
+    val byteArray = BigInt(s, 16).toByteArray
+    if (byteArray.length > 1 && byteArray(0) == 0) {
+      // remove sign byte for positive numbers if exists
+      byteArray.tail
+    } else {
+      byteArray
+    }
+  }
+
+  test("listagg function") {
```

Comment:
should we put the SQL tests in the golden file?

Reply:
Yes, but here there are the same concerns about non-deterministic nature we are discussing above with @MaxGekk. And if in tests I can eventually add something like 
```Scala
      checkAnswer(
        sql("select string_agg(b) from df group by a"),
        Row(null) :: Row(OneOf("bc", "cb")) :: Row(OneOf("cd", "dc")) :: Nil)
``` 
in golden files it wouldn't be so easy.

But again this problem always has been (since 2021) so probably it's not a problem and I am too pedantic 😖  
https://github.com/apache/spark/blob/b0414b19a0ea5ffa10067154c00009259419f045/sql/core/src/test/resources/sql-tests/results/group-by.sql.out#L1062-L1071